reclaimer/meta/wrappers/src/byteswapping_ext.c

```diff
@@ -0,0 +1,144 @@
+    } else if (field_offset_bufs[1].itemsize != 4) {
+        PySys_FormatStdout("four_byte_offs array format must be 'I'.\n");
+    } else if (field_offset_bufs[2].itemsize != 4) {
+        PySys_FormatStdout("eight_byte_offs array format must be 'I'.\n");
+    } else if (end > bufs[0].len) {
+        PySys_FormatStdout("end must within bounds of input buffer.\n");
+    } else if (bufs[0].buf == bufs[1].buf) {
+        PySys_FormatStdout("src and dst cannot be the same.\n");
+    } else if (field_offset_bufs[2].itemsize != 4) {
+        PySys_FormatStdout("eight_byte_offs array format must be 'I'.\n");
+    } else if (struct_size == 0) {
+        PySys_FormatStdout("struct size must be greater than zero.\n");
+    } else if (start > end) {
+        PySys_FormatStdout("start must be less than end.\n");
+    } else {
+        int item_count;
+        for (int i = 0; i < 3; i++) {
+            if (field_offset_bufs[i].buf == NULL)
+                continue;
+
+            item_count = field_offset_bufs[i].len / field_offset_bufs[i].itemsize;
+            for (int j = 0; j < item_count; j++)
+                if (((int *)field_offset_bufs[i].buf)[j] + field_offset_sizes[i] > struct_size) {
+                    PySys_FormatStdout("offets must be within the struct size.\n");
+                    goto exit;
+                }
+        }
+        // everything looks good. lets process!
+        item_count = field_offset_bufs[0].len / field_offset_bufs[0].itemsize;
+        if (field_offset_bufs[0].buf != NULL && item_count > 0)
+            byteswap_two_byte_array(
+                bufs[0].buf, bufs[1].buf, start, end, struct_size,
+                field_offset_bufs[0].buf, item_count
+            );
+
+        item_count = field_offset_bufs[1].len / field_offset_bufs[1].itemsize;
+        if (field_offset_bufs[1].buf != NULL && item_count > 0)
+            byteswap_four_byte_array(
+                bufs[0].buf, bufs[1].buf, start, end, struct_size,
+                field_offset_bufs[1].buf, item_count
+            );
+
+        item_count = field_offset_bufs[2].len / field_offset_bufs[2].itemsize;
+        if (field_offset_bufs[2].buf != NULL && item_count > 0)
+            byteswap_eight_byte_array(
+                bufs[0].buf, bufs[1].buf, start, end, struct_size,
+                field_offset_bufs[2].buf, item_count
+            );
+    }
+
+    exit:
+    // Release the buffer objects
+    RELEASE_PY_BUFFER_ARRAY(bufs)
+    RELEASE_PY_BUFFER_ARRAY(field_offset_bufs)
+
+    return Py_BuildValue("");  // return Py_None while incrementing it
+}
+
+/* A list of all the methods defined by this module.
+"METH_VARGS" tells Python how to call the handler.
+The {NULL, NULL} entry indicates the end of the method definitions */
+static PyMethodDef byteswapping_ext_methods[] = {
+    {"byteswap_struct_array", py_byteswap_struct_array, METH_VARARGS, ""},
+    {NULL, NULL, 0, NULL}      /* sentinel */
```

Comment:
Terminator TANTAN TAN TANTAN, TANTAN TAN TANTAN :robot:
Something saying like `/* Terminate list*/` would be more descriptive imo.

Reply:
This is boilerplate copypasta'd from arbytmap and the sounds module. If we're gonna change it, we should change them all. Consistency is important.